### PR TITLE
making TS happen

### PR DESCRIPTION
**Pull request by filipbech**
I changed the outDir so the build-files are in the dist-folder and created a index.js in the root that re-exports everything from the dist-file...

Then I added some any-definitions to shut the ts-compiler up - we could look into what the correct types should be, but I'm not exactly sure for some of these cases. 

Then I added a comment with an idea, and one with what I think is a bug, but lets talk about them another time.

### CODE REVIEW

File: src/lit-element.ts [hunk 1]
```diff
@@ -69,13 +69,14 @@ export class LitElement extends HTMLElement {
...
+    //FBB: idea -> should we call this with `this` so you can use destructuring assignment (like preact does with state and props)
     return html``;
   }
   
   attributeChangedCallback(prop: string, _oldValue: string, newValue: string) {
-    const { type: typeFn } = this.constructor.properties[prop];
-
+    const { type: typeFn } = (this.constructor as any).properties[prop];
     if (typeFn.name === 'Boolean') {
+      //FBB: I believe there is a bug here!
```
Comment (kenchris) [inline src/lit-element.ts]: below or above? :-)

File: src/lit-element.ts [hunk 2]
```diff
@@ -45,9 +45,9 @@ export class LitElement extends HTMLElement {
     for (const prop in this.properties) {
       const { type: typeFn, value, attrName } = this.properties[prop];
 
-      Object.defineProperty(this.prototype, prop, {
+      Object.defineProperty(this.prototype, prop, <any>{
         get() { return this._values[prop] || value; },
-        set(v) {
+        set(v:any) {
```
Comment (kenchris) [inline src/lit-element.ts]: I have some proper fixes for these, so please rebase after I land them

File: lib/lit-element.d.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { TemplateResult } from '../node_modules/lit-html/lit-html.js';
```
Comment (kenchris) [inline lib/lit-element.d.ts]: I don't think these files should be checked in - they are generated

File: index.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export * from './lib/lit-element.js'
```
Comment (kenchris) [inline index.js]: You need to update the examples in docs/

Comment (kenchris) [reply index.js]: You can test with "polymer serve" or similar

**Comment by kenchris** [comment]
OK, I redid your patch and pushed, and I believe that I incorporated all the correct changes.

Could you check to see if I have done it correctly, or if not, rebase this patch?

File: src/lit-element.ts (hunk 1 above)
Comment (filipbech) [reply src/lit-element.ts]: below... if you remove the attribute the property value is supposed go to false right?

File: lib/lit-element.d.ts (hunk 3 above)
Comment (filipbech) [reply lib/lit-element.d.ts]: we need to for npm right?

**Comment by filipbech** [comment]
I believe what we need to change the import path to the lit-html, so it doesn't assume a local node_modules folder... so instead of '../node_modules/lit-html' I guess it should be ../../lit-html ?

File: src/lit-element.ts (hunk 1 above)
Comment (kenchris) [reply src/lit-element.ts]: You are right... should be === prop or === "". When removing it becomes null

**Comment by filipbech** [comment]
Lets just close this PR

**Pull request closed without merging**